crates/communication/src/server/parameters/directory.rs

```diff
@@ -0,0 +1,240 @@
+
+    let location_default_file_path = location_directory.join("default.json");
+    if location_default_file_path.exists() {
+        let location_default_parameters = from_path(location_default_file_path)
+            .await
+            .map_err(DirectoryError::DefaultParametersOfLocationNotGet)?;
+        merge_json(&mut parameters, &location_default_parameters);
+    }
+
+    let body_file_path = root_path.as_ref().join(format!("body.{}.json", body_id));
+    if body_file_path.exists() {
+        let body_parameters = from_path(body_file_path)
+            .await
+            .map_err(DirectoryError::BodyParametersNotGet)?;
+        merge_json(&mut parameters, &body_parameters);
+    }
+
+    let head_file_path = root_path.as_ref().join(format!("head.{}.json", head_id));
+    if head_file_path.exists() {
+        let head_parameters = from_path(head_file_path)
+            .await
+            .map_err(DirectoryError::HeadParametersNotGet)?;
+        merge_json(&mut parameters, &head_parameters);
+    }
+
+    let location_body_file_path = location_directory.join(format!("body.{}.json", body_id));
+    if location_body_file_path.exists() {
+        let location_body_parameters = from_path(location_body_file_path)
+            .await
+            .map_err(DirectoryError::BodyParametersOfLocationNotGet)?;
+        merge_json(&mut parameters, &location_body_parameters);
+    }
+
+    let location_head_file_path = location_directory.join(format!("head.{}.json", head_id));
+    if location_head_file_path.exists() {
+        let location_head_parameters = from_path(location_head_file_path)
+            .await
+            .map_err(DirectoryError::HeadParametersOfLocationNotGet)?;
+        merge_json(&mut parameters, &location_head_parameters);
+    }
+
+    from_value(parameters).map_err(DirectoryError::JsonValueNotConvertedToParameters)
+}
+
+pub async fn serialize<Parameters>(
+    parameters: &Parameters,
+    root_path: impl AsRef<Path>,
+    body_id: &str,
+    head_id: &str,
+) -> Result<(), DirectoryError>
+where
+    Parameters: Serialize,
+{
+    let mut parameters =
+        to_value(parameters).map_err(DirectoryError::ParametersNotConvertedToJsonValue)?;
+    let stored_parameters = to_value(deserialize(&root_path, body_id, head_id).await?)
+        .map_err(DirectoryError::ParametersNotConvertedToJsonValue)?;
+
+    prune_equal_branches(&mut parameters, &stored_parameters);
+
+    let location_head_file_path = root_path
+        .as_ref()
+        .join(get_location_directory(head_id))
+        .join(format!("head.{}.json", head_id));
+
+    let mut location_head_parameters = from_path(&location_head_file_path)
+        .await
+        .map_err(DirectoryError::BodyParametersOfLocationNotGet)?;
+    merge_json(&mut location_head_parameters, &parameters);
+
+    to_path(location_head_file_path, location_head_parameters)
+        .await
+        .map_err(DirectoryError::HeadParametersOfLocationNotSet)
+}
```

Comment:
This currently cannot remove items that already exist in the location head configuration.

I thought that could be solved by just switching the order of pruning and merging, but since the location head parameters are also considered when loading the stored parameters, they would always be equal (unless overridden by the parameters in memory) and thus always pruned.

We'd have to load all layers of parameters except for the last one to get proper values to compare against.
That may be outside the scope for this PR, feel free to implement or resolve.

Reply:
I don't get why you want to remove items in the first place. It sounds like you are trying to chase a artificial use-case. The intention of the current implementation is to keep the location head configuration and add the changed items to it. Which scenario do you have in mind, where this does not work? 🤔

Reply:
- start with default config
- change some value
- save to disk
- change mind, change value back
- save to disk again

If I understand correctly, the key that was changed now still persists in the location head configuration even though it is equal to the value in the rest of the configuration files.

Reply:
I see. Yes, this is true. We discussed this yesterday that this would require a lot more intelligence or more control from the client-side. IMHO out-of-scope. I will add a bullet point to Ideas for next iteration.